core/src/task.rs

```diff
@@ -533,48 +537,151 @@ macro_rules! vdaf_dispatch {
     };
 }
 
-/// An authentication (bearer) token used by aggregators for aggregator-to-aggregator and
-/// collector-to-aggregator authentication.
-#[derive(Clone)]
-pub struct AuthenticationToken(Vec<u8>);
+/// Different modes of authentication supported by Janus for either sending requests (e.g., leader
+/// to helper) or receiving them (e.g., collector to leader).
+#[derive(Clone, Derivative, Serialize, Deserialize)]
+#[derivative(Debug)]
+#[serde(tag = "type", content = "token")]
+#[non_exhaustive]
+pub enum AuthenticationToken {
+    /// A bearer token. The value is an opaque byte string. Its Base64 encoding is inserted into
+    /// HTTP requests as specified in [RFC 6750 section 2.1][1]. The token is not necessarily an
+    /// OAuth token.
+    ///
+    /// [1]: https://datatracker.ietf.org/doc/html/rfc6750#section-2.1
+    Bearer(
+        #[derivative(Debug = "ignore")]
+        #[serde(
+            serialize_with = "as_url_safe_unpadded_base64",
+            deserialize_with = "from_url_safe_unpadded_base64"
+        )]
+        Vec<u8>,
+    ),
+
+    /// Token presented as the value of the "DAP-Auth-Token" HTTP header. Conforms to
+    /// [draft-ietf-dap-ppm-01 section 3.2][1].
+    ///
+    /// [1]: https://datatracker.ietf.org/doc/html/draft-ietf-ppm-dap-01#name-https-sender-authentication
+    DapAuth(DapAuthToken),
+}
 
 impl AuthenticationToken {
-    /// Constructs a bearer token string suitable for use as the value in an HTTP `Authorization`
-    /// header.
-    pub fn bearer_token(&self) -> String {
-        format!("Bearer {}", STANDARD.encode(self.as_ref()))
+    /// Returns an HTTP header and value that should be used to authenticate an HTTP request with
+    /// this credential.
+    pub fn request_authentication(&self) -> (&'static str, Vec<u8>) {
```

Comment:
Ideally this would be expressed as something like an extension trait on `reqwest::RequestBuilder` with a method `RequestBuilder::with_authentication(token: &AuthenticationToken)`, but we need to use these values in `mockito` and `trillium` as well, hence this somewhat less ergonomic, lowest-common-denominator approach.